flang/lib/Lower/OpenMP.cpp

```diff
@@ -125,6 +125,189 @@ createBodyOfOp(Op &op, Fortran::lower::AbstractConverter &converter,
     privatiseVars(converter, *clauses);
 }
 
+static std::int64_t
+createOrderedDependSourceOp(Fortran::lower::AbstractConverter &converter,
+                            Fortran::lower::pft::Evaluation &eval,
+                            SmallVectorImpl<Value> &operands) {
+  auto &firOpBuilder = converter.getFirOpBuilder();
+  auto currentLocation = converter.getCurrentLocation();
+  // The number of associated loops in the doacross loop nest.
+  std::int64_t orderedLevel{0};
+
+  auto evalIter = eval.parentConstruct;
+  while (evalIter) {
+    if ((*evalIter).isDirective()) {
+      const auto ompConstruct =
+          (*evalIter).getIf<Fortran::parser::OpenMPConstruct>();
+      if (const auto &loopConstruct{
+              std::get_if<Fortran::parser::OpenMPLoopConstruct>(
+                  &ompConstruct->u)}) {
+        const auto &loopClauseList = std::get<Fortran::parser::OmpClauseList>(
+            std::get<Fortran::parser::OmpBeginLoopDirective>(loopConstruct->t)
+                .t);
+        // Get the value of the parameter specified in the ORDERED clause in the
+        // worksharing-loop directive.
+        for (const auto &clause : loopClauseList.v) {
+          if (const auto &orderedClause{
+                  std::get_if<Fortran::parser::OmpClause::Ordered>(
+                      &clause.u)}) {
+            orderedLevel = *Fortran::evaluate::ToInt64(
+                Fortran::semantics::GetExpr(orderedClause->v));
+            break;
+          }
+        }
+        if (orderedLevel > 0) {
+          // Get the loop iterators in the doacross loop nest and create the
+          // indexes of the loop iterators to form the depend clause operands.
+          const auto &doConstruct =
+              std::get<std::optional<Fortran::parser::DoConstruct>>(
+                  loopConstruct->t);
+          const Fortran::parser::DoConstruct *loop{&*doConstruct};
+          std::int64_t orderedLevelInd{0};
+          while (orderedLevelInd < orderedLevel) {
+            const auto &name = std::get<Fortran::parser::LoopControl::Bounds>(
+                                   loop->GetLoopControl()->u)
+                                   .name.thing;
+            auto varSym = converter.getSymbolAddress(*name.symbol);
+            mlir::Value ConstantOne = firOpBuilder.createIntegerConstant(
+                currentLocation, firOpBuilder.getI64Type(), 1);
+            mlir::Value varI32, varI64, varIndex;
```

Comment:
Why not declare these where/when they are defined?

Reply:
Fixed.